pkg/registry/deployment/strategy.go

```diff
@@ -92,6 +92,8 @@ func (deploymentStatusStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newDeployment := obj.(*extensions.Deployment)
 	oldDeployment := old.(*extensions.Deployment)
 	newDeployment.Spec = oldDeployment.Spec
+	newDeployment.Labels = oldDeployment.Labels
```

Comment:
@bgrant0607 , @janetkuo is right, this is status update. I don't think it happens elsewhere.